src/valkey-cli.c

```diff
@@ -2226,9 +2228,29 @@ static int cliReadReply(int output_raw_strings) {
         fflush(stdout);
         sdsfree(out);
     }
+    
     return REDIS_OK;
 }
 
+/* Helper method to process unsubscribe responses */
+static void processUnsubscribeReply(redisReply *reply) {
+    if (reply->elements >= 3) {
+        char *cmd = reply->element[0]->str;
+        int count = reply->element[2]->integer;
+ 
+        if (strcmp(cmd, "unsubscribe") == 0 || strcmp(cmd, "punsubscribe") == 0) {
+            config.pubsub_unsharded_count = count;
+        } else if (strcmp(cmd, "sunsubscribe") == 0) {
+            config.pubsub_sharded_count = count;
+        }
+
+        if (config.pubsub_unsharded_count == 0 && config.pubsub_sharded_count == 0) {
+            config.pubsub_mode = 0;
+            cliRefreshPrompt();
+        }
```

Comment:
Why not let this helper handle subscribe and unsubscribe messages together here? They have almost the same logic. (We can name it handlePubsubMode or something.)

It seems to me that we can call this after just checking `isPubsubPush` and then we can remove any other code for setting the pubsub mode, right?

```suggestion
        if (strcmp(cmd, "subscribe") == 0 || strcmp(cmd, "psubscribe") == 0 ||
            strcmp(cmd, "unsubscribe") == 0 || strcmp(cmd, "punsubscribe") == 0) {
            config.pubsub_unsharded_count = count;
        } else if (strcmp(cmd, "ssubscribe") == 0 || strcmp(cmd, "sunsubscribe") == 0) {
            config.pubsub_sharded_count = count;
        }

        if (config.pubsub_unsharded_count == 0 && config.pubsub_sharded_count == 0 && config.pubsub_mode) {
            config.pubsub_mode = 0;
            cliRefreshPrompt();
        } else if (config.pubsub_unsharded_count + config.pubsub_sharded_count > 0 && !config.pubsub_mode) {
            config.pubsub_mode = 1;
            cliRefreshPrompt();
        }
```